### ISSUE TITLE
Typescript-class component toast message in modal usage

### ISSUE THREAD

**Issue by tarikpnr** [calintamas/react-native-toast-message#172]
![issue](https://user-images.githubusercontent.com/61876765/113987416-0ea1e600-9857-11eb-8290-4d5bc0878eb2.png)

Hello. I wanted to use toast message in a modal. I initialized the steps related to showing in modal as shown in docs. But having problem while creatingRef. What should be the proper way to use toast message in typescript- class component ?

The error is shown in screenshot above. Any help would be great. Thanks...

**Comment by tarikpnr** [comment]
Other attempt...


![issue2](https://user-images.githubusercontent.com/61876765/113988247-f2eb0f80-9857-11eb-9bbf-dffe3acabc46.png)
...

**Comment by calintamas** [comment]
Hi @tarikpnr , try creating a ref in the class constructor using [React.createRef()](https://reactjs.org/docs/react-api.html#reactcreateref)

```js
constructor(props) {
  super(props);
  this.modalRef = React.createRef();
}
```

**Comment by tarikpnr** [comment]
![Screenshot_1](https://user-images.githubusercontent.com/61876765/114282882-4bd0c880-9a4f-11eb-852c-d5e5fb614a6a.png)

Im guessing in order to access the functions of Toast I need to specify a type to it. Currently if I try to use toast message on modal `private readonly modalRef:any` having any type (workaround)...
![Screenshot_2](https://user-images.githubusercontent.com/61876765/114282986-ffd25380-9a4f-11eb-8631-b48a542edb70.png)



still I don't get toast message on the modal. It shows up on top of the screen for a split-second, then it disappears.
![Screenshot_3](https://user-images.githubusercontent.com/61876765/114283031-432cc200-9a50-11eb-9351-9064cf4eed90.png)

**Comment by tarikpnr** [comment]
![Screenshot_4](https://user-images.githubusercontent.com/61876765/114283352-fcd86280-9a51-11eb-9b4d-4d7212613d52.png)

When I specify zIndex of both Modal and Toast component, the toast message shows up. But I am not able to hide it by pressing on it. 

![Screenshot_5](https://user-images.githubusercontent.com/61876765/114283375-19749a80-9a52-11eb-984a-0e80ab72c25f.png)

**Comment by calintamas** [comment]
@tarikpnr to hide it, you need to call `hide` on the same ref
```js
this.modalRef.current.show({
  ...
  onPress: () => this.modalRef.current.hide()
})
```

As for creating the ref, you can set it to be of `Toast` type:
```js
this.modalRef = React.createRef<Toast>();
```

**Comment by tarikpnr** [comment]
`this.modalRef.current.hide()` that did the trick! Thanks. 

But on the other side, while setting type to `React.createRef()`, still getting error below.

![Screenshot_1](https://user-images.githubusercontent.com/61876765/114301238-aeb47500-9acc-11eb-8f42-1968c7be3573.png)


![Screenshot_3](https://user-images.githubusercontent.com/61876765/114301340-fe933c00-9acc-11eb-9eec-b14ced46b566.png)

**Comment by jailsonpaca** [comment]
1+, I am with the same problem

**Comment by Onibenjo** [comment]
To fix the typescript error, i just created a variable and in the ref, added
```js
// doing this autocompletion, could be set to null
let modalToastRef  = Toast

// then set the variable to the ref
<Toast ref={ref => (modalToastRef = ref)} />
```

**Comment by calintamas** [comment]
Fixed in `v2.0.0`. [Read the complete changelog](https://github.com/calintamas/react-native-toast-message/releases/tag/v2.0.0). 

If you find any issues with the v2 implementation, feel free to reopen this issue. Thanks!
